### PR TITLE
Namespace all the classes

### PR DESCRIPTION
**Pull request by gordon81**
Converted the first nine classes

Tx_Solr_Access_Rootline  -> ApacheSolrForTypo3\Solr\Access\Rootline  
Tx_Solr_Access_RootlineElementFormatException -> ApacheSolrForTypo3\Solr\Access\RootlineElementFormatException
Tx_Solr_Access_RootlineElement -> ApacheSolrForTypo3\Solr\Access\RootlineElement
Tx_Solr_Backend_ContentElementWizardIconProvider -> ApacheSolrForTypo3\Solr\Backend\ContentElementWizardIconProvider
Tx_Solr_Backend_IndexingConfigurationSelectorField -> ApacheSolrForTypo3\Solr\Backend\IndexingConfigurationSelectorField
Tx_Solr_Cli_Dispatcher -> ApacheSolrForTypo3\Solr\Cli\Dispatcher
Tx_Solr_ContentObject_Content -> ApacheSolrForTypo3\Solr\ContentObject\Content
Tx_Solr_ContentObject_Multivalue -> ApacheSolrForTypo3\Solr\ContentObject\Multivalue
Tx_Solr_ContentObject_Relation -> ApacheSolrForTypo3\Solr\ContentObject\Relation

### CODE REVIEW

File: Classes/Access/Rootline.php [hunk 1]
```diff
@@ -86,14 +87,13 @@ class Tx_Solr_Access_Rootline {
 	public function __construct($accessRootline = NULL) {
 		if (!is_null($accessRootline)) {
 			$rawRootlineElements = explode(self::ELEMENT_DELIMITER, $accessRootline);
-
 			foreach ($rawRootlineElements as $rawRootlineElement) {
 				try {
-					$this->push(t3lib_div::makeInstance(
-						'Tx_Solr_Access_RootlineElement',
+					$this->push(\TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance(
```
Comment (irnnr) [inline Classes/Access/Rootline.php]: Please don't mix unrelated changes, it makes reviews unnecessarily harder. Would you mind reverting these kind of changes? We can/will do this in separate commits.

**Comment by irnnr** [comment]
Please use meaningful titles for the commits. Right now they're all "Namespace all the classes", which does not allow to see what classes are affected by which commit.

**Comment by irnnr** [comment]
First of all, thanks for your help, it's highly appreciated! There are a few things we should pay attention to, however:

Please do not add unrelated changes. All the replacements for the t3lib classes can (and should!) be done separately. Currently it is very hard to review your changes.

Also, as mentioned in the issue (#30), stick to one PR per folder. This also helps making reviews easier and allows others to contribute, too without introducing many merge conflicts.

File: Classes/Access/Rootline.php (hunk 1 above)
Comment (gordon81) [inline Classes/Access/Rootline.php]: then I have to do so \t3lib_div

**Comment by gordon81** [comment]
Ok look tomorrow as I have to make the adjustments in PR 

Everything is new for me with github (sorry)

File: Classes/Access/Rootline.php (hunk 1 above)
Comment (irnnr) [inline Classes/Access/Rootline.php]: that's alright

**Comment by irnnr** [comment]
no problem, we're all here to learn and started somewhere :)

**Comment by irnnr** [comment]
let me know if you need any help / have any questions!

**Comment by irnnr** [comment]
Please rebase and force push your PR branch so that old comments collapse

**Comment by gordon81** [comment]
done rebase

**Comment by irnnr** [comment]
Thanks Gordon! 

Do you now also see why I wanted you to do a force push? It collapses/removes all the old comments for the old commits :)

File: ext_autoload.php [hunk 2]
```diff
@@ -188,12 +188,12 @@
 
 		// scheduler tasks
 
-	'tx_solr_scheduler_indexqueueworkertask' => $extensionPath . 'Scheduler/IndexQueueWorkerTask.php',
-	'tx_solr_scheduler_indexqueueworkertaskadditionalfieldprovider' => $extensionPath . 'Scheduler/IndexQueueWorkerTaskAdditionalFieldProvider.php',
-	'tx_solr_scheduler_reindextask' => $extensionPath . 'Scheduler/ReIndexTask.php',
-	'tx_solr_scheduler_reindextaskadditionalfieldprovider' => $extensionPath . 'Scheduler/ReIndexTaskAdditionalFieldProvider.php',
+	'tx_solr_piresults_parameterkeepingformmodifier' => $extensionPath . 'PiResults/ParameterKeepingFormModifier.php',
```
Comment (irnnr) [inline ext_autoload.php]: Not sure what's going on here, guessing this just slipped in. Can you revert this chunk please?

L191-L194

**Comment by irnnr** [comment]
There's only that one change in ext_autoload.php that needs to be fixed. I'd then merge this PR as a first step. Great job!

File: Classes/Typo3PageIndexer.php [hunk 3]
```diff
@@ -22,9 +22,6 @@
 *  This copyright notice MUST APPEAR in all copies of the script!
 ***************************************************************/
 
-use TYPO3\CMS\Core\Utility\GeneralUtility;
```
Comment (irnnr) [inline Classes/Typo3PageIndexer.php]: Please revert this change

File: Classes/ContentObject/Relation.php [hunk 4]
```diff
@@ -161,7 +161,7 @@ protected function getRelatedItemsFromForeignTable($localFieldName, $localRecord
 				$whereClause = $foreignTableName . '.uid = ' . (int) array_shift($foreignTableUids);
 			}
 		}
-		$pageSelector = GeneralUtility::makeInstance('TYPO3\\CMS\\Frontend\\Page\\PageRepository');
+		$pageSelector = GeneralUtility::makeInstance('t3lib_pageSelect');
```
Comment (irnnr) [inline Classes/ContentObject/Relation.php]: This reverts a previous change

File: Classes/Typo3PageIndexer.php [hunk 5]
```diff
@@ -23,8 +23,6 @@
 ***************************************************************/
 
 use TYPO3\CMS\Core\Utility\GeneralUtility;
-use TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController;
```
Comment (irnnr) [inline Classes/Typo3PageIndexer.php]: This reverts a previous change

File: Classes/Typo3PageIndexer.php [hunk 6]
```diff
@@ -510,7 +508,7 @@ public function getDocumentsSentToSolr() {
 	 */
 	protected function getDocumentIdGroups() {
 		$groups = $this->pageAccessRootline->getGroups();
-		$groups = Tx_Solr_Access_Rootline::cleanGroupArray($groups);
+		$groups = \ApacheSolrForTypo3\Solr\Access\Rootline::cleanGroupArray($groups);
```
Comment (irnnr) [inline Classes/Typo3PageIndexer.php]: please feel free to import the classes

**Comment by irnnr** [comment]
👍🏻👏🏻
